src/main/java/de/tum/in/test/api/util/ClassMemberAccessor.java

```diff
@@ -0,0 +1,160 @@
+	 * <p>
+	 * Also recursively searches for the field in all superclasses and implemented
+	 * interfaces.
+	 *
+	 * @param declaringClass The class that declares or inherits the field.
+	 * @param fieldName      The name of the attribute.
+	 * @param findPrivate    True, if this method should search for (package)
+	 *                       private or protected (inherited) attributes.
+	 * @return The wanted field.
+	 * @throws NoSuchFieldException Thrown if the specified field cannot be found.
+	 */
+	static Field getAttribute(Class<?> declaringClass, String fieldName, boolean findPrivate)
+			throws NoSuchFieldException {
+		return getClassHierarchy(declaringClass).flatMap(c -> {
+			try {
+				if (c.equals(declaringClass) || findPrivate) {
+					// Also search for declared fields in the own class even when not explicitly
+					// searching for private fields to be able to provide error messages to the
+					// users that the field exists, but with the wrong visibility.
+					return getInheritedField(declaringClass, c, fieldName).stream();
+				} else {
+					return Stream.of(c.getField(fieldName));
+				}
+			} catch (NoSuchFieldException nsfe) {
+				return Stream.empty();
+			}
+		}).findFirst().orElseThrow(() -> new NoSuchFieldException(fieldName));
+	}
+
+	/**
+	 * Searches for a field in the target class that might be inherited from the
+	 * declaring class.
+	 *
+	 * @param targetClass    The class in which the field should be accessible.
+	 * @param declaringClass The class from which the field might be inherited.
+	 * @param fieldName      The name of the attribute.
+	 * @return A field that is accessible in the target class.
+	 * @throws NoSuchFieldException Thrown if no field with the given name could be
+	 *                              found in either class.
+	 */
+	private static Optional<Field> getInheritedField(Class<?> targetClass, Class<?> declaringClass, String fieldName)
+			throws NoSuchFieldException {
+		Field field = declaringClass.getDeclaredField(fieldName);
+		if (isInheritable(targetClass, declaringClass, field.getModifiers())) {
+			return Optional.of(field);
+		} else {
+			return Optional.empty();
+		}
+	}
+
+	/**
+	 * Searches for all classes and interfaces the given class is inheriting from.
+	 * <p>
+	 * The given class itself is part of the result. Performs a depth-first search
+	 * starting with classes, then interfaces.
+	 *
+	 * @param clazz The class for which the inheritance tree should be traversed.
+	 * @return A stream of the given class and all superclasses and interfaces it
+	 *         inherits from.
+	 */
+	private static Stream<Class<?>> getClassHierarchy(Class<?> clazz) {
+		Stream<Class<?>> directSuperclasses = Stream
+				.concat(Stream.of(clazz.getSuperclass()), Arrays.stream(clazz.getInterfaces()))
+				.filter(Objects::nonNull);
+		return Stream.concat(Stream.of(clazz), directSuperclasses.flatMap(ClassMemberAccessor::getClassHierarchy));
+	}
```

Comment:
Looks good to me:

Given
```java
X extends Y
B extends A implements Z
C extends B implements X
```
we get
```haskell
-> C
(C B X)
(C (B A Z) X)
(C (B (A Object) Z) X)
(C (B (A (Object)) Z) X)
(C (B (A (Object)) (Z)) X)
(C (B (A (Object)) (Z)) (X Y))
(C (B (A (Object)) (Z)) (X (Y)))
-> C B A Object Z X Y
```
which complies with the Java implementation.